tools/testoperator/src/tests/http/consistency.rs

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use crate::{
+    commands::HttpConsistencyTestArgs,
+    tests::{get_app_and_start_request, util::Color},
+    with_color,
+};
+use std::{sync::Arc, time::Duration};
+use test_framework::{
+    anyhow::{self, anyhow},
+    arrow::array::ArrowNativeTypeOp,
+    metrics::MetricCollector,
+    spiced::SpicedInstance,
+    spicetest::{
+        http::consistency::{self, ConsistencyConfig},
+        SpiceTest,
+    },
+};
+
+/// Runs a test to ensure the P50 & p90 latencies do not increase by some threshold over the
+/// duration of the test when N clients are sending queries concurrently.
+pub async fn consistency_run(args: &HttpConsistencyTestArgs) -> anyhow::Result<()> {
+    let (_app, start_request) = get_app_and_start_request(&args.http.common, None)?;
+    let component = args.http.get_http_component()?;
+    let payloads: Vec<_> = args
+        .http
+        .get_payloads()?
+        .into_iter()
+        .map(Arc::from)
+        .collect();
+
+    let mut spiced_instance = SpicedInstance::start(start_request).await?;
+
+    spiced_instance
+        .wait_for_ready(Duration::from_secs(args.http.common.ready_wait))
+        .await?;
+
+    let test = SpiceTest::new(
+        "Consistency Test".to_string(),
+        spiced_instance,
+        consistency::NotStarted::new(ConsistencyConfig::new(
+            Duration::from_secs(args.http.common.duration),
+            args.http.common.concurrency,
+            payloads,
+            component,
+            args.buckets,
+        )),
+    );
+
+    let results = test.start()?.wait().await?.metrics()?;
```

Comment:
```suggestion
    let test = test.start()?.wait().await?;
    let results = test.metrics()?;
    let mut spiced_instance = test.end();
    spiced_instance.stop()?;
```

we should retain control over the spiced instance instead of letting it `Drop`